Homework4: Yuliya Golovach

YuliyaGolovach opened the pull request.

YuliyaGolovach committed:
Fix of a code style warnings.

YuliyaGolovach committed:
Adjust line lengths to conform code style.

YuliyaGolovach committed:
Some whitespaces added, import order fixed.

YuliyaGolovach committed:
Back to origin commit in master branch

YuliyaGolovach committed:
Add five tasks

YuliyaGolovach committed:
Add the rest of tasks

YuliyaGolovach committed:
Add new lines at the end of some files

YuliyaGolovach committed:
Fix flake8 errors

YuliyaGolovach committed:
Add 'the biggest number' task from lection

YuliyaGolovach committed:
Rename files

Rename files according to python_style case. Translate russian comments into english.

belovmd commented on src/hw_4/t05_pupils_and_languages.py:
```diff
@@ -0,0 +1,68 @@
+print(len(popular_languages))
+for language in popular_languages:
+    print(language)
+print(len(non_popular_languages))
```
pupils_list already contain it, isn't it?

belovmd commented on src/hw_4/t08_biggest_number_from_lection.py:
```diff
@@ -0,0 +1,18 @@
+""" The biggest number
```
great

belovmd reviewed (changes requested):
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-python-spring/blob/master/homework_review/Homework4.md):
- 1
- 2

belovmd commented on src/hw_4/t03_different_numbers.py:
```diff
@@ -0,0 +1,27 @@
+
+set1 = set(input().split())
+set2 = set(input().split())
+print(len(set1.intersection(set2)))
```
it counts numbers in both lists and doesn't solve task description

YuliyaGolovach committed:
Address review note #1

Change task 2 solution (cities and countries) to address review note #1. Use defaultdict as dictionary of sets. Add reading from txt file as more convenient way of input.

YuliyaGolovach committed:
Change solution for task #3

Use symmetric difference.

YuliyaGolovach committed:
Address review note #2

Change solution to use intersection and union of sets.

belovmd reviewed (approved):
9,5
